tests/unit/components/sl-date-range-picker-test.js

```diff
@@ -0,0 +1,352 @@
+
+    assert.equal(
+        component.get( 'earliestEndDate' ),
+        '01/01/2015'
+    );
+});
+
+test( 'Latest start date is the based on max date and end date', function( assert ) {
+    let component = this.subject();
+
+    assert.strictEqual(
+        component.get( 'latestStartDate' ),
+        null
+    );
+
+    component.set( 'maxDate', '01/01/2029' );
+
+    assert.equal(
+        component.get( 'latestStartDate' ),
+        '01/01/2029'
+    );
+
+    component.set( 'endDateValue', '01/01/2015' );
+
+    assert.equal(
+        component.get( 'latestStartDate' ),
+        '01/01/2015'
+    );
+});
+
+test( 'Events from start date input are removed upon willClearRender', function( assert ) {
+    let component = this.subject();
+    let startDateInput = this.$( '.sl-daterange-start-date input' )[0];
+
+    assert.equal(
+        Ember.typeOf( $._data( startDateInput, 'events' ).change ),
+        'array',
+        'Start date input has change event listener after render'
+    );
+
+    component.trigger( 'willClearRender' );
+
+    assert.strictEqual(
+        $._data( startDateInput, 'events' ),
+        undefined,
+        'Start date input has no event listeners after willClearRender'
+    );
+});
+
+test( 'label, startDatePlaceholder, and endDatePlaceholder are undefined by default', function( assert ) {
+    let component = this.subject();
+
+    assert.strictEqual(
+        component.get( 'label' ),
+        undefined,
+        'label is undefined by default'
+    );
+
+    assert.strictEqual(
+        this.$( 'label' )[0],
+        undefined,
+        'No label is created when label is undefined'
+    );
+
+    assert.strictEqual(
+        component.get( 'startDatePlaceholder' ),
+        undefined,
+        'startDatePlaceholder is undefined by default'
+    );
+
+    assert.strictEqual(
+        this.$( '.sl-daterange-start-date input' ).prop( 'placeholder' ),
+        '',
+        'Start date input placeholder is empty when startDatePlaceholder is undefined'
+    );
+
+    assert.strictEqual(
+        component.get( 'endDatePlaceholder' ),
+        undefined,
+        'endDatePlaceholder is undefined by default'
+    );
+
+    assert.strictEqual(
+        this.$( '.sl-daterange-end-date input' ).prop( 'placeholder' ),
+        '',
+        'End date input placeholder is empty when endDatePlaceholder is undefined'
+    );
+});
+
+test( 'label is accepted as a parameter', function( assert ) {
+    let labelText = 'lorem ipsum';
+    let component = this.subject({ label: labelText });
+
+    assert.equal(
+        component.get( 'label' ),
+        labelText,
+        'label was accepted as a parameter'
+    );
+
+    assert.equal(
+        this.$( 'label' ).html(),
+        labelText,
+        'label element was created with label parameter text'
+    );
+
+    assert.equal(
+        this.$( 'label' ).prop( 'for' ),
+        component.get( 'inputElementId' ),
+        'label element has the correct for property'
+    );

```

Comment:
Prefer this test to get the id of the first date picker DOM element and compare it to the `for` property of the `label` property.  It's an "outside of things" way to ensure that everything is wired up correctly and that we're not getting the wrong value.  If the value was being set incorrectly, this assertion would still pass, when what we're really wanting to test is that the label of this component is for the id of the first date picker component rendered in the DOM.